scripts/earthengine/events_pipeline.py

```diff
@@ -0,0 +1,818 @@
+            'skip_existing_output':
+                True,
+        },
+
+        # Settings to convert source input data to csv
+        # See raster_to_csv.py:_DEFAULT_CONFIG for config settings
+        {
+            'stage':
+                'raster_csv',
+            'skip_existing_output':
+                True,
+            's2_level':
+                10,
+            'output_file':
+                'gs://{gcs_bucket}/{gcs_folder}/input_csv/{year}/{import_name}-{stage_name}-{start_date}-{time_period}.csv'
+        },
+        # Settings to process data into events
+        # See process_events.py:_DEFAULT_CONFIG for config settings.
+        {
+            'stage':
+                'events',
+            'output_dir':
+                'gs://{gcs_bucket}/{gcs_folder}/events/{year}/data.{file_extension}'
+        },
+    ],
+
+    # File with the latest state for the pipeline
+    'pipeline_state_file': ''
+}
+
+# Processing state persisted to a file
+# Includes dynamic parameter values for each stage of processing.
+_PROCESS_STATE = {
+    # Last date for which input was processed in YYYY-MM-DD format
+    'last_input_date': '',
+}
+
+_DEBUG = False
+
+
+class StageRunner:
+    '''Class to run a single stage of the pipeline.
+    Create a derived class with run() method calling necessary processing.
+    Example:
+      class MyStage(StageRunner):
+        def __init__(self,
+                     config_dicts: list = [],
+                     state: dict = {},
+                     counters=None):
+          self.setUp('My-Stage', configs, state, counters)
+
+        def run(self,
+            input_files: list = None,
+            config: dict = {},
+            counters: Counters = None) -> list:
+          outputs = []
+          for file in input_files:
+            outputs.append(my_process(file, config, counters))
+          return outputs
+
+    '''
+
+    def __init__(self,
+                 config_dicts: list = [],
+                 state: dict = {},
+                 counters=None):
+        self.setUp('runner', config_dicts, state, counters)
+
+    def run(self,
+            input_files: list = None,
+            config: dict = {},
+            counters: Counters = None) -> list:
+        '''Override in child class.
+       Returns the output files generated after running processing for this stage
+       on the input_files.
+
+       Args:
+         input_files: List of files to process
+           they are outputs from previous stage along with any files
+           matching config['input_files'].
+          config: dictionary of configuration settings for this stage.
+          counters: global counters for the task.
+       Returns:
+         list if output files that are passed on to the next stage as inputs.
+        '''
+        logging.fatal(
+            f'run() not implemented in {self.__class__}. Use a class with run() overridden.'
+        )
+        return []
+
+    def setUp(self,
```

Comment:
nit: why setUp and not set_up :)

Reply:
was using convention similar to unittest.textCase.setUp()
changed to set_up()